### ISSUE TITLE
xml.onload does not act on if(no success)

### ISSUE THREAD

**Issue by delta04** [ruffle-rs/ruffle#2787]
The xml.onLoad-function  seems to ignore the ” if(success)”, when it is run in ruffle.
myXml.onLoad = function(success:Boolean) {
 	 if (success) {   	
	"execute   function A();"
	} else {
       " execute   function B();"
 	}
}		
Ruffle reads success every time, and executes function A() - even if the xml-file doesn’t exist.
The attached files show what the problem is. There is both a ruffle- and a flash – html to show the difference,  - plus the ruffle-folder, the xml-file and the fla. (They are all located in the same folder on the site)

[xml.onload_not_working_properly.zip](https://github.com/ruffle-rs/ruffle/files/5859949/xml.onload_not_working_properly.zip)

**Comment by EmperorBale** [comment]
This is probably because ruffle is not checking the status of the http request, it seems to just check if it sent back data, making ruffle check what status code the server returned would probably fix this issue

**Comment by delta04** [comment]
If the xml-file doesn't exist, it reads the error message below in ruffle, - whereas it returns nothing in flash - as it should. ( I'm not sure though, if this is helpful at all)
![error message](https://user-images.githubusercontent.com/41504455/105631176-74342800-5e4d-11eb-86e1-3ecf6e6ab7a7.png)

**Comment by delta04** [comment]
If the xml-file doesn't exist, it reads the error message below in ruffle,
- whereas it returns nothing in flash - as it should. ( I'm not sure
though, if this is helpful at all) [image: error message.png]


Den lør. 23. jan. 2021 kl. 20.30 skrev EmperorBale <notifications@github.com

**Comment by delta04** [comment]
If the xml-file doesn't exist, it reads the error message below in ruffle, - whereas it returns nothing in flash - as it should. ( I'm not sure though, if this is helpful at all)
![error message](https://user-images.githubusercontent.com/41504455/105641614-3ce16d80-5e85-11eb-8de5-ea6c0e0f5269.png)

**Comment by delta04** [comment]
I found a solution by changing the flash file. I moved the if-condition out of the xml-loader.
The information that the xml-loader had to act on, was if-file-exist or not, - which it couldn't. I realized that this information is available already when the exercise book is opened. If a saved file exists it will load the students saved work, if not it will leave an (unseen) information "undefined". And that can be used to choose between two xml-loaders, when the student presses "save" later in the process.
In other words:   saveBtn.onPress=  ....  
If(startInfo==undefined) ... play.xml-loader_1// (which reads the saved file, and updates it)
 ..... else .... play.xml-loader_2 (which fetches a sort of standard-file, updates that and saves it)
